encrypted-media/scripts/syntax-mediakeysession.js

```diff
@@ -0,0 +1,489 @@
+        // tests is the same.
+
+        // Too few parameters.
+        {
+            exception: 'TypeError',
+            func: function (mk1, type) {
+                return mk1.createSession().generateRequest(type);
+            }
+        },
+        // Invalid parameters.
+        {
+            exception: 'TypeError',
+            func: function (mk2, type) {
+                return mk2.createSession().generateRequest(type, '');
+            }
+        },
+        {
+            exception: 'TypeError',
+            func: function (mk3, type) {
+                return mk3.createSession().generateRequest(type, null);
+            }
+        },
+        {
+            exception: 'TypeError',
+            func: function (mk4, type) {
+                return mk4.createSession().generateRequest(type, undefined);
+            }
+        },
+        {
+            exception: 'TypeError',
+            func: function (mk5, type) {
+                return mk5.createSession().generateRequest(type, 1);
+            }
+        },
+        {
+            exception: 'InvalidAccessError',
+            func: function (mk6, type) {
+                return mk6.createSession().generateRequest(type, new Uint8Array(0));
+            }
+        }
+    ];
+    function generateRequestTestExceptions(){
+        return new Promise(function(resolve, reject){
+            isInitDataTypeSupported(keysystem, initDataType)
+                .then(function (isTypeSupported) {
+                    assert_true(isTypeSupported, "initDataType not supported");
+                    return navigator.requestMediaKeySystemAccess(keysystem, [configuration]);
+                })
+                .then(function (access) {
+                    return access.createMediaKeys();
+                })
+                .then(function (mediaKeys) {
+                    var mp4SessionPromises = kTypeSpecificGenerateRequestExceptionsTestCases.map(function (testCase) {
+                        return test_exception(testCase, mediaKeys, initDataType, initData);
+                    });
+                    assert_not_equals(mp4SessionPromises.length, 0);
+                    return Promise.all(mp4SessionPromises);
+                })
+                .then(function (result) {
+                    resolve()
+                })
+                .catch(function (error) {
+                    reject(error);
+                });
+        })
+    }
+    promise_test(function(){
+        return generateRequestTestExceptions();
+    }, testname + ' test MediaKeySession generateRequest() exceptions.')
+
+    var kLoadExceptionsTestCases = [
+        // Too few parameters.
+        {
+            exception: 'TypeError',
+            func: function (mk1) {
+                return mk1.createSession('temporary').load();
+            }
+        },
+        // 'temporary' sessions are never allowed, so always return
+        // 'InvalidAccessError'.
+        {
+            exception: 'InvalidAccessError',
```

Comment:
InvalidAccessError should be TypeError for non persistent sessions (and for persistent sessions with invalid sessionIds).